app/views/template_inputs/_invocation_form.html.erb

```diff
@@ -1,7 +1,12 @@
 <%= input_values_fields.fields_for input.id.to_s, composer.input_value_for(input) do |input_fields| %>
-  <% unless input.options.blank? %>
+  <% if input.options.present? %>
     <%= selectable_f input_fields, :value, input.options_array, { :include_blank => !input.required }, :label_help => input.description, :require => input.required, :label => input.name, :id => input.name, :onchange => local_assigns[:onchange] %>
   <% else %>
-    <%= textarea_f input_fields, :value, :label => input.name, :label_help => input.description, :required => input.required, :rows => 2, :onchange => local_assigns[:onchange], :id => input.name %>
+    <%= content_tag :div, nil, id: "template-input-#{input.name}" %>
+    <% if input.value_type == 'plain' || input.value_type.nil? %>
+      <%= textarea_f input_fields, :value, :label => input.name, :label_help => input.description, :required => input.required, :rows => 2, :onchange => local_assigns[:onchange], :id => input.name %>
+    <% else %>
+      <%= mount_report_template_input(input) %>
```

Comment:
It didnt get into the review for some reason, so:

We can not retrieve errors from an input. We could though do this:
```suggestion
      <%= mount_report_template_input(f.object) %>
```
which gives us a `template_value`.
And than in the helper we can:
```ruby
def mount_report_template_input(input_value)
  input = input_value.template_input
  # ...
  initialError: input_value.errors.full_messages.join("\n").presence,
  # ...
```

Reply:
as @Laviro mentioned, the `initialError` prop, is under develop, a follow up PR would come soon, right @Laviro ? :)